src/main/java/com/google/changestreams/sample/bigquery/changelog/fullschema/SchemaUtils.java

```diff
@@ -0,0 +1,539 @@
+    "_metadata_spanner_commit_timestamp";
+  public static final String BQ_CHANGELOG_FIELD_NAME_SERVER_TRANSACTION_ID =
+    "_metadata_spanner_server_transaction_id";
+  public static final String BQ_CHANGELOG_FIELD_NAME_RECORD_SEQUENCE =
+    "_metadata_spanner_record_sequence";
+  public static final String BQ_CHANGELOG_FIELD_NAME_IS_LAST_RECORD_IN_TRANSACTION_IN_PARTITION =
+    "_metadata_spanner_is_last_record_in_transaction_in_partition";
+  public static final String BQ_CHANGELOG_FIELD_NAME_NUMBER_OF_RECORDS_IN_TRANSACTION =
+    "_metadata_spanner_number_of_records_in_transaction";
+  public static final String BQ_CHANGELOG_FIELD_NAME_NUMBER_OF_PARTITIONS_IN_TRANSACTION =
+    "_metadata_spanner_number_of_partitions_in_transaction";
+  public static final String BQ_CHANGELOG_FIELD_NAME_BIGQUERY_COMMIT_TIMESTAMP =
+    "_metadata_big_query_commit_timestamp";
+
+  private static final String INFORMATION_SCHEMA_TABLE_NAME = "TABLE_NAME";
+  private static final String INFORMATION_SCHEMA_COLUMN_NAME = "COLUMN_NAME";
+  private static final String INFORMATION_SCHEMA_SPANNER_TYPE = "SPANNER_TYPE";
+  private static final String INFORMATION_SCHEMA_ORDINAL_POSITION = "ORDINAL_POSITION";
+  private static final String INFORMATION_SCHEMA_CONSTRAINT_NAME = "CONSTRAINT_NAME";
+  private static final String INFORMATION_SCHEMA_ALL = "ALL";
+
+  /**
+   * @return the fields that are only used intermediately in the pipeline and are not corresponding
+   * to Spanner columns.
+   */
+  public static Set<String> getBigQueryIntermediateMetadataFieldNames() {
+    Set<String> fieldNames = new HashSet<>();
+    fieldNames.add(BQ_CHANGELOG_FIELD_NAME_ERROR);
+    fieldNames.add(BQ_CHANGELOG_FIELD_NAME_RETRY_COUNT);
+    fieldNames.add(BQ_CHANGELOG_FIELD_NAME_ORIGINAL_PAYLOAD_JSON);
+    return fieldNames;
+  }
+
+  /**
+   * @return a map where the key is the table name and the value is the SpannerTable object of the
+   * table name.
+   */
+  public static Map<String, SpannerTable> getSpannerTableByName(
+    DatabaseClient databaseClient, String changeStreamName) {
+    final Set<String> spannerTableNames = getSpannerTableNamesTrackedByChangeStreams(
+      databaseClient, changeStreamName);
+
+    final Map<String, Set<String>> spannerColumnNamesExplicitlyTrackedByChangeStreamByTableName =
+      getSpannerColumnNamesExplicitlyTrackedByChangeStreamsByTableName(
+        databaseClient, changeStreamName);
+
+    return getSpannerTableByName(databaseClient, spannerTableNames,
+      spannerColumnNamesExplicitlyTrackedByChangeStreamByTableName);
+  }
+
+  private static void appendWhereTableNamesEqualToSql(
+    StringBuilder stringBuilder, Set<String> spannerTableNames) {
+    if (!spannerTableNames.isEmpty()) {
+      stringBuilder.append(" WHERE");
+
+      for (String tableName : spannerTableNames) {
+        stringBuilder.append(" TABLE_NAME=\"");
+        stringBuilder.append(tableName);
```

Comment:
This should be in a parameterized query `... WHERE TABLE_NAME IN @array`

Reply:
Did you mean "WHERE TABLE IN ('t1', 't2')"? It doesn't seem we can use array after "IN".